core/src/main/resources/org/apache/spark/ui/static/historypage.js

```diff
@@ -38,8 +38,17 @@ function makeIdNumeric(id) {
 }
 
 function formatDate(date) {
-  if (date <= 0) return "-";
-  else return date.split(".")[0].replace("T", " ");
+  if (date <= 0) {
+    return "-";
+  } else {
+    var dt = new Date(date);
+    return dt.getFullYear() + "-" +
+      ("0" + (dt.getMonth() + 1)).slice(-2) + "-" +
```

Comment:
this looks a bit hacky, is there any way in js to print a number to 2 letters?

Reply:
How about this:
```javascript
function formatDate(date) {
  if (date <= 0) {
    return "-";
  } else {
    var dt = new Date(date);
    return dt.getFullYear() + "-" +
      (dt.getMonth() + 1).toString().padStart(2, "0") + "-" +
      (dt.getDate()).toString().padStart(2, "0") + " " +
      (dt.getHours()).toString().padStart(2, "0") + ":" +
      (dt.getMinutes()).toString().padStart(2, "0") + ":" +
      (dt.getSeconds()).toString().padStart(2, "0");
  }
}
```

Reply:
yea much better, `.toString().padStart(2, "0")` can be a method.